Src/java/cql-to-elm/src/main/java/org/cqframework/cql/cql2elm/LibraryManager.java

```diff
@@ -328,11 +328,10 @@ private TranslatedLibrary generateTranslatedLibrary(Library library) {
                 for (ExpressionDef expressionDef : library.getStatements().getDef()) {
 
                     //to do implement an ElmTypeInferencingVisitor; make sure that the resultType is set for each node
-                    if(!(expressionDef.getExpression() instanceof  SingletonFrom)) {
+                    if (!(expressionDef.getExpression() instanceof SingletonFrom)) {
                         if (expressionDef.getResultTypeSpecifier() != null || expressionDef.getResultType() != null) {
```

Comment:
And it needs to be looking at the `resultType` property, not the `resultTypeSpecifier` or `resultTypeName`. ELM has classes for representings types in the TypeSpecifier hierarchy, and if the type specifier is only a name, ELM elements can use a QName attribute directly, but the actual `resultType` element is not in ELM (since it;s a graph, and has references to in-memory objects). The `resultType` property is introduced on the Trackable base class, and the translator uses that to store the results of type inference as it goes, and then ensures that the type there is output as a typeSpecifier in the ELM. So in general, a freshly deserialized ELM library will not have `resultType` set, and cannot be used directly by the translator.

Reply:
So basically, in the absence of an implementation of the TypeInferencingVisitor, we can't use deserialized ELM in the translator. So this test should fail, and when we get that in place we can add a passing test.